### ISSUE TITLE
Wrong property names when using QueryMultiple with dynamic types

### ISSUE THREAD

**Issue by mark64digital** [CollaboratingPlatypus/PetaPoco#355]
When using QueryMultiple with dynamic types and the select queries have the same number of columns the second result set uses the same property names as the first but the correct data.

PetaPoco.Core 5.1.228

using (var results = db.QueryMultiple(@"

    SELECT id, name, email
    FROM customers

    SELECT id, customerId, address
    FROM addresses

"))
{
    var customers = results.Read<dynamic>().ToList();
    var addresses = results.Read<dynamic>().ToList();

    // customers
    // id   name    email
    // 1    mark    mark@emai.com

    // addresses
    // id   name    email
    // 101  1       1 high street
}

**Comment by maniish** [comment]
This is true. We are also facing the same issue and it would be great if it resolved asap.

**Comment by VickyMahir** [comment]
We are also facing same issue.

**Comment by VickyMahir** [comment]
I have found there is performance issue with QueryMultiple.

When we have data size is 12.1 KB then its take 25 to 27 second to fetch.

I think this cause because of dynamic data structure.

So please check performance for this.

Thanks,
Vijay M

**Comment by t0mburton** [comment]
Hi,
I also came across this issue and found that this is due to cache key being the objects type, which is obviously the same in this scenario (System.Object) so the properties are not repopulated.
Is dynamic supported in this library as [this](https://github.com/CollaboratingPlatypus/PetaPoco/blob/development/PetaPoco/Core/PocoData.cs#L102) seems to suggest it isn't? The dynamic type does not map to System.Dynamic.ExpandoObject so maybe I'm wrong with this assumption, apologies if so.

I'm happy to submit a PR to correct this issue, I'm just after some guidance on which way you would like to proceed with this.

Thanks.

**Comment by pleb** [comment]
Hi @t0mburton, dynamic is supported. However, I'm not sure if it's supported everywhere. Feel free to submit a PR. There's also a project with integration tests where you can add your tests.

**Comment by angtianqiang** [comment]
我也遇到同样的问题，不知道怎么解决，可以方案？
